lib/sedna/service/client.py

```diff
@@ -151,39 +151,34 @@ def __init__(self, url, client_id, **kwargs):
 
     async def connect(self):
         LOGGER.info(f"{self.uri} connection by {self.client_id}")
-
-        try:
-            self.ws = await asyncio.wait_for(websockets.connect(
-                self.uri, **self.kwargs
-            ), self._ws_timeout)
-            await self.ws.send(json.dumps({'type': 'subscribe',
-                                           'client_id': self.client_id}))
-
-            res = await self.ws.recv()
-            return res
-        except ConnectionRefusedError:
-            LOGGER.info(f"{self.uri} connection was refused by server")
-            raise
-        except ConnectionClosedError:
-            LOGGER.info(f"{self.uri} connection lost")
-            raise
-        except ConnectionClosedOK:
-            LOGGER.info(f"{self.uri} connection closed")
-            raise
-        except InvalidStatusCode as err:
-            LOGGER.info(
-                f"{self.uri} websocket failed - "
-                f"with invalid status code {err.status_code}")
-            raise
-        except WebSocketException as err:
-            LOGGER.info(f"{self.uri} websocket failed - with {err}")
-            raise
-        except OSError as err:
-            LOGGER.info(f"{self.uri} connection failed - with {err}")
-            raise
-        except Exception:
-            LOGGER.exception(f"{self.uri} websocket Error")
-            raise
+        for _ in range(self._retry):
```

Comment:
I note you have already use `retry` package, you can use it here again.

Reply:
fixed. As `retry` no longer maintained, I'll replace it with `tenacity`